content/docs/languages-sdks/yaml/yaml-language-reference.md

```diff
@@ -116,23 +116,34 @@ There are at least two reasons to explicitly define providers in YAML, or explic
 1. Using explicit versions enables pinning the dependencies used, a technique used to improve build reliability.
 2. Using explicit providers enables controlling the options for providers used by each resource, as described in [Unlock Programmatic Control by Disabling Default Providers](/blog/disable-default-providers/).
 
-#### Resource version
+If a specific provider version is necessary, we prefer the approaches in the order described here.
+
+#### Default provider
+
+Declaring a default provider simplifies configuring a provider version across all resources for that provider, as all subsequent resources will 
 
-To create a resource with a specific provider version use the `version` option as described in [Resource Options](#resource-options):
+To configure the default provider instance for all resources for that provider's name, use the [`resources`](#resources) section to declare a provider resource.
+
+For the `type` property, prefix the name of the provider with `pulumi:providers`, for example, `pulumi:providers:aws`. Setting the `defaultProvider` property to `true`, defined at the same level as `type`, configures this as a default provider.
 
 ```yaml
 resources:
-  something:
-    type: aws:s3:Bucket
-    properties:
-      ...
+  aws-provider:
+    type: pulumi:providers:aws
+    defaultProvider: true
     options:
-      version: 5.6.0
+      version: 6.45.0
+  my-bucket:
+    type: aws:s3:Bucket
 ```
 
+In this example the `my-bucket` resource, and any other AWS resources, uses the default AWS provider version 6.45.0.
+
+Only one provider of a given `type` can be declared as the default provider.
+
 #### Explicit provider
 
-To create an explicit provider instance, preferably with a specific version, use the [`resources`](#resources) section. For the `type` property, prefix the name of the provider with `pulumi:providers`.
+To declare an explicit provider instance for specific resources, declare a provider resource the [`resources`](#resources) section. For the `type` property, prefix the name of the provider with `pulumi:providers`.
```

Comment:
Bit of a nit fixed here, Pulumi programs don't create resources, they declare them.